src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -813,10 +813,20 @@ private void CompileSynthesizedExplicitImplementations(SourceMemberContainerType
             // we are not generating any observable diagnostics here so it is ok to short-circuit on global errors.
             if (!_globalHasErrors)
             {
+                var interfaces = sourceTypeSymbol.GetInterfacesToEmit();
+
                 var discardedDiagnostics = BindingDiagnosticBag.GetInstance(_diagnostics);
                 foreach (var synthesizedExplicitImpl in sourceTypeSymbol.GetSynthesizedExplicitImplementations(_cancellationToken).ForwardingMethods)
                 {
                     Debug.Assert(synthesizedExplicitImpl.SynthesizesLoweredBoundBody);
+
+                    // Avoid emitting duplicate forwarding methods (e.g., when the class implements the same interface twice with different nullability).
+                    if (!interfaces.Contains(synthesizedExplicitImpl.ExplicitInterfaceImplementations[0].ContainingType,
+                        Symbols.SymbolEqualityComparer.ConsiderEverything))
+                    {
+                        continue;
```

Comment:
>continue; [](http://example.com/codeflow?start=24&length=9)

It would be good to explain why it is safe to skip this method, i.e. why do we expect to find an exact match for each implemented method.

Reply:
Both sets of interfaces we are comparing (`GetInterfacesToEmit` and the containing types of `ForwardingMethods`) are based on `NamedTypeSymbol.InterfacesNoUseSiteDiagnostics`. The former set is deduplicated via CLRSignature comparer, the latter set contains all interfaces. So, the former is a subset of the latter, i.e., we must reach all methods of the former (`GetInterfacesToEmit`).

Reply:
Consider adding this as a comment here and in other two similar places